File: dotnet/DesignPatterns/BehavioralDesignPatterns/ChainOfResponsibilityDesignPattern/ChainOfResponsibilityDesignPattern.Example2/Handlers/FileTypeValidatorHandler.cs
namespace ChainOfResponsibilityDesignPattern.Example2.Handlers;

public class FileTypeValidatorHandler : BaseFileHandler
{
    private readonly List<string> _allowedExtensions = new() 
    { 
        ".pdf", ".doc", ".docx", ".txt", ".jpg", ".jpeg", ".png", ".gif" 
    };

    public override void Handle(FileUploadRequest request)
    {
        if (!_allowedExtensions.Contains(request.FileExtension.ToLower()))
        {
            request.IsValid = false;
            request.ValidationMessages.Add($"File type '{request.FileExtension}' is not allowed. Allowed: {string.Join(", ", _allowedExtensions)}");
            return; // Stop the chain
        }
        
        // Pass to next handler
        base.Handle(request);
    }
}

File: nodejs/DesignPatterns/BehavioralDesignPatterns/ChainOfResponsibilityDesignPattern/ChainOfResponsibilityDesignPattern.Example2/src/Handlers/FileTypeValidatorHandler.ts
import { FileUploadRequest } from "../FileUploadRequest";
import { BaseFileHandler } from "../BaseFileHandler";

const ALLOWED_EXTENSIONS = [".pdf", ".doc", ".docx", ".txt", ".jpg", ".jpeg", ".png", ".gif"];

export class FileTypeValidatorHandler extends BaseFileHandler {
  handle(request: FileUploadRequest): void {
    if (!ALLOWED_EXTENSIONS.includes(request.fileExtension.toLowerCase())) {
      request.isValid = false;
      request.validationMessages.push(
        `File type '${request.fileExtension}' is not allowed. Allowed: ${ALLOWED_EXTENSIONS.join(", ")}`
      );
      return;
    }
    super.handle(request);
  }
}
